CRM/Event/Form/Registration.php

```diff
@@ -1598,15 +1597,17 @@ protected function validatePriceSet(array $params, $priceSetId, $priceSetDetails
    * @return mixed|null
    */
   public function getSubmittedValue(string $fieldName) {
-    $value = parent::getSubmittedValue($fieldName);
-    // Check for value as well in case the field has been added to the Confirm form
-    // I don't quite know how that works but something Matt has worked on.
-    if ($value || !in_array($this->getName(), ['Confirm', 'ThankYou'], TRUE)) {
-      return $value;
-    }
-    // If we are on the Confirm or ThankYou page then the submitted values
-    // were on the Register Page so we return them
-    $value = $this->controller->exportValue('Register', $fieldName);
+    if ($this->isShowPaymentOnConfirm() && in_array($this->getName(), ['Confirm', 'ThankYou'], TRUE)) {
+      $value = $this->controller->exportValue('Confirm', $fieldName);
```

Comment:
We'll find payment fields on Confirm if payment was made on Confirm. Otherwise they'll be on Register